### PR TITLE
Add support for usual reporters feature

### LINKED ISSUE #23: Add a "usual_reporters" field in the project configuration so the bot reacts with the project emoji
It would be useful for deployments of Hebbot with many known projects to have the possibility to add a `usual_reporters` to individual projects, as follows:

```
[[projects]]
emoji = '📻'
name = 'shortwave'
title = 'Shortwave'
description = 'Internet radio player with over 25000 stations.'
website = 'https://gitlab.gnome.org/World/Shortwave'
default_section = 'circle'
usual_reporters = [
    '@felix:homeserver.net',
    '@thib:serverhome.org'
]
```

In the example above, if either Felix or Thib report news to Hebbot, it will react with the 📻 emoji. The editors then only need to click the emoji to sort it in the Shortwave project.

If a same reporter is listed in several projects' `usual_reporters` list, the bot reacts with the emojis of each projects the reporter is listed in, and the reporter can then click on the one actually related to the report.

### PR DESCRIPTION
**Pull request by thibaultamartin**
Closes #23

### CODE REVIEW

File: src/bot.rs [hunk 1]
```diff
@@ -148,6 +148,17 @@ impl Bot {
             .expect("Unable to send message");
     }
 
+    /// Simplified method for sending a reaction emoji
+    async fn send_reaction(&self, reaction: &str, event_id: &EventId) {
```
Comment (haecker-felix) [inline src/bot.rs:152]: Should rename `event_id` to something more obvious like `msg_event_id`

File: src/bot.rs [hunk 2]
```diff
@@ -148,6 +148,17 @@ impl Bot {
...
+        let content = ReactionEventContent::new(Relation::new(event_id.clone(), reaction.to_string()));
+        let content = AnyMessageEventContent::Reaction(content);
+        //println!("I should add the {} emoji eventually!", reaction);
```
Comment (haecker-felix) [inline src/bot.rs:155]: That should be removed ;)

File: src/bot.rs [hunk 3]
```diff
@@ -272,7 +283,7 @@ impl EventCallback {
             return;
         }
 
-        let event_id = event_id.to_string();
+        //let event_id = event_id.to_string();
```
Comment (haecker-felix) [inline src/bot.rs:286]: why?

**Review by haecker-felix — CHANGES_REQUESTED** [review]
We should special case that the bot ignores his own reaction events for the case someone adds the bot itself as editor.

File: src/bot.rs (hunk 1 above)
Comment (thibaultamartin) [reply src/bot.rs]: Fixed in eacffc6

File: src/bot.rs (hunk 2 above)
Comment (thibaultamartin) [reply src/bot.rs]: Fixed in eacffc6

File: src/bot.rs (hunk 3 above)
Comment (thibaultamartin) [reply src/bot.rs]: I need the event ID (the EventID struct) a few lines below. This overrode the parameter and I couldn't retrieve it :)

**Comment by thibaultamartin** [comment]
> We should special case that the bot ignores his own reaction events for the case someone adds the bot itself as editor.

Fixed in 5b395c3

File: configuration/config.toml.example [hunk 4]
```diff
@@ -31,11 +31,16 @@ title = 'Shortwave'
 description = 'Internet radio player with over 25000 stations.'
 website = 'https://gitlab.gnome.org/World/Shortwave'
 default_section = 'circle'
+editors = [
```
Comment (haecker-felix) [inline configuration/config.toml.example:34]: Is it `usual_reporters` or `editors`?

File: configuration/config.toml.example [hunk 5]
```diff
@@ -31,11 +31,16 @@ title = 'Shortwave'
...
 [[projects]]
 emoji = '⬇️'
 name = 'fragments'
 title = 'Fragments'
 description = 'Easy to use BitTorrent client.'
 website = 'https://gitlab.gnome.org/World/Fragments'
-default_section = 'circle'
\ No newline at end of file
+default_section = 'circle'
+editors = [] # Mandatory for empty projects
```
Comment (haecker-felix) [inline configuration/config.toml.example:46]: No need to add mandatory info, every config field is mandatory.

File: src/bot.rs [hunk 6]
```diff
@@ -271,7 +282,6 @@ impl EventCallback {
             return;
         }
 
-        let event_id = event_id.to_string();
```
Comment (haecker-felix) [inline src/bot.rs:274]: Keep this, then you don't need to convert it in line 309.

**Review by haecker-felix — CHANGES_REQUESTED** [review]

**Comment by haecker-felix** [comment]
I wonder if we should do the same thing for sections as well. For example in TWIG we have a reporter who often reports stuff for a section (eg. GNOME Circle section). So it would be neat if hebbot automatically suggests section emojis too.

File: configuration/config.toml.example (hunk 4 above)
Comment (thibaultamartin) [reply configuration/config.toml.example]: Fxied in 358e8fe4f998d4460201044ed7315e70e7404e61

File: configuration/config.toml.example (hunk 5 above)
Comment (thibaultamartin) [reply configuration/config.toml.example]: Fixed in 358e8fe4f998d4460201044ed7315e70e7404e61

**Comment by thibaultamartin** [comment]
> I wonder if we should do the same thing for sections as well. For example in TWIG we have a reporter who often reports stuff for a section (eg. GNOME Circle section). So it would be neat if hebbot automatically suggests section emojis too.

I added support for sections as well

**Pull request merged**